### PR TITLE
Optimisations of Monster::Rand

### PR DESCRIPTION
**Pull request by bjauny**
Created a static map to associate a `level_t` to a `monster_t`. In the case of a random monster the level is randomized, like before.

`Rand` now only creates an instance of `Monster` if it is actually needed. The little overhead of allocating space each time due to not having the call to `reserve` anymore can be ignored I think.

### CODE REVIEW

**Comment by bjauny** [comment]
No error when built on my station, but C++98 rejected the way the static map was initialized. I created a private function to do the initialization, should be good now.

**Comment by bjauny** [comment]
Fixes #57

**Comment by ihhub** [comment]
@bjauny thanks for the pull request! I'll take a look at it soon.

File: src/fheroes2/monster/monster.cpp [hunk 1]
```diff
@@ -177,6 +177,8 @@ StreamBase & operator>> (StreamBase & msg, MonsterStaticData & obj)
     return msg;
 }
 
+std::map<Monster::monster_t, Monster::level_t> Monster::monsterLevel = InitializeMonsterLevels();
```
[the remark this answers is not in the document]
Comment (bjauny) [reply src/fheroes2/monster/monster.cpp]: Done

File: src/fheroes2/monster/monster.h [hunk 2]
```diff
@@ -115,6 +115,8 @@ class Monster
 	MONSTER_RND
     };
 
+    static std::map<monster_t, level_t> monsterLevel;
```
[the remark this answers is not in the document]
Comment (bjauny) [reply src/fheroes2/monster/monster.h]: Done

File: src/fheroes2/monster/monster.h [hunk 3]
```diff
@@ -183,11 +184,94 @@ class Monster
...
     static void UpdateStats(const std::string &);
     static float GetUpgradeRatio(void);
+    static level_t GetLevel(const monster_t);
 
 protected:
     static Monster FromDwelling(int race, u32 dw);
 
     int id;
+private:
+    static std::map<monster_t, level_t> InitializeMonsterLevels() {
```
[the remark this answers is not in the document]
Comment (bjauny) [reply src/fheroes2/monster/monster.h]: Done

**Comment by andrey-starodubtsev** [comment]
This PR cannot be built on both *nix and Win platforms; it also has no activity for almost 2 months.

Closed in favor of #245.

**Pull request closed without merging**